#BN-32 feat: Reset Password

Lydie260 opened the pull request:
.Reset Password
Ticket Trello Link
https://trello.com/c/7MuiGSjf/32-reset-password


# PR Description
This PR Allows User to reset Password in Case they forgot it by receiving a reset email with link and click on it and be able to reset the password and also be able to log in with the new password.
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] Send reset Email to the user with reset link.
- [x] User click the link and be able to reset Password.
- [x] User Enters new Password
- [x] User Log In with new password
# Description of tasks that were expected to be completed and that are not yet completed
# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`

# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

uwdjany reviewed (commented):
resolve the conflicts

Honore123 reviewed (dismissed):
🚀

uwdjany reviewed (dismissed):
well done

Honore123 reviewed (dismissed):
🚀

dushimeemma reviewed (commented):
Kindly write the unit test for your endpoints @Lydie260 , otherwise good job!

dushimeemma commented on .github/workflows/gitActions.yml:
```diff
@@ -0,0 +1,31 @@
+    - name: Run linter
+      run: npm run lint
+    - name: Run tests
+      run: npm test
```
add a empty line at the end of this file

dushimeemma commented on src/docs/ResetPasswordDocs.js:
```diff
@@ -0,0 +1,185 @@
+ *                   type: string
+ *                   description: Error message
+ *                   example: Invalid email/password
+ */
```
add an empty line at the end of this file

dushimeemma commented on src/index.js:
```diff
@@ -47,7 +47,7 @@ app.use(googleAuth);
 
 app.get('/', (req, res) => res.status(200).json({ status: 200, message: req.t('welcome_message') }));
-
+app.use('/ResetPasswordDocs', swaggerUi.serve, swaggerUi.setup(specs));
```
why have you changed this line?

dushimeemma commented on src/index.js:
```diff
@@ -59,3 +59,10 @@ app.listen(port, async () => {
   console.log(`app listening on port ${port}`, process.env.NODE_ENV);
 });
 export default app;
+
```
live only one empty line at the end of the file

dushimeemma commented on src/routes/API/SendResetEmail.js:
```diff
@@ -0,0 +1,69 @@
+  // Update user's password
+  User.resetToken= null
+  User.resetTokenExpiry=null
+  User.password = newPassword
```
why don't you hash the password?

dushimeemma commented on src/routes/API/SendResetEmail.js:
```diff
@@ -0,0 +1,69 @@
+const express = require('express');
```
why are you mixing commonjs and es6?

dushimeemma commented on src/routes/API/SendResetEmail.js:
```diff
@@ -0,0 +1,69 @@
+
+
+
+module.exports = router;
```
use es6 syntax

dushimeemma commented on src/tests/sum.test.js:
```diff
@@ -0,0 +1,19 @@
+const chai = require('chai');
```
- why are you testing the sum?
- why are you using commonjs syntax?

Honore123 reviewed (dismissed):
🚀

Niyobuzima reviewed (dismissed):
**Well Done**

Honore123 reviewed (dismissed):
🚀

kunda4 committed:
BN-16 feat: A seller should be able to delete a specific item (#54)

* * feat(logout):create the logout feature
-included user logout
-documented the user logout
[finishes #30]

* feat(recommended product): worked on an algorithm to generate product recommendation to the buyer (#46)

- created a function in recommendedProdcutController for a single product view which will save all the transaction in the of viewd prodcut by the buyer
- created product recommendation algorithm that checks the buyer wishlist and viewed products and categories to generates the recommended products, it also include similar buyer preferences.
- worked on the api swagger documentation of the above function
- worked on the internationalization of responsesworked on test cases for the above function
[Finishes #27]

---------

Co-authored-by: Nelly KAWERA <78723561+nellykawera@users.noreply.github.com>
Co-authored-by: Bertin NDAHAYO SINGIZWA <100227921+Illustre13@users.noreply.github.com>

kunda4 reviewed (approved):
Well done

Niyobuzima reviewed (dismissed):
**well done**

kunda4 reviewed (dismissed):
Good job

dushimeemma commented on src/config/config.json:
I am not sure the reason why this file is here

dushimeemma commented on src/controllers/loginController.js:
Please use prettier extension to format your code, if you are using vs code.

Niyobuzima reviewed (dismissed):
Nice 👍

Niyobuzima reviewed (dismissed):
**Done**

dushimeemma commented on src/routes/API/SendResetEmail.js:
```diff
@@ -0,0 +1,72 @@
+const router = express.Router();
+import { use } from 'chai';
+import {user} from '../../database/models';
+const { sendPasswordResetEmail } = require('../../utils/email');
```
This is commonJS, kindly use ES6

dushimeemma commented on src/routes/API/SendResetEmail.js:
```diff
@@ -0,0 +1,72 @@
+import { use } from 'chai';
+import {user} from '../../database/models';
+const { sendPasswordResetEmail } = require('../../utils/email');
+const crypto = require('crypto');
```
This is commonJS kindly use ES6

dushimeemma commented on src/routes/API/SendResetEmail.js:
```diff
@@ -0,0 +1,72 @@
+
+
+
+module.exports = router;
```
This is commonJS syntax kindly use ES6

dushimeemma commented on src/routes/routes.js:
```diff
@@ -37,6 +34,8 @@ import wishListRouter from './wishListRouter';
 import recommendedProduct from './recommendedProductRoute';
 
 // Login route
+const authRoutes = require('./API/SendResetEmail');
```
This is commonJS syntax kindly use ES6

dushimeemma commented on src/utils/email.js:
```diff
@@ -0,0 +1,25 @@
+    subject: 'Password Reset',
+    text: `You are receiving this email because you (or someone else) have requested a password reset for your account.\n\n
+      Please click on the following link or paste it into your browser to complete the process:\n\n
+      http://localhost:9090/auth/reset/${resetToken}\n\n
```
Kindly change your email and use `dotenv variables` this will help on using different values for development, staging and production.

Niyobuzima commented on src/database/models/index.js:
```diff
@@ -34,3 +34,5 @@ Object.keys(db).forEach((modelName) => {
 db.sequelize = sequelize;
 db.Sequelize = Sequelize;
```
is this line necessary

Lydie260 replied:
yes very necessary because resetToken is needed and I saw in some codes you remove it please do not remove it again because once I push and you pull and delete it my resetPassword endpoint won't work

Niyobuzima commented on src/routes/API/SendResetEmail.js:
```diff
@@ -0,0 +1,76 @@
+
+export default router;
+
+module.exports = router;
```
delete this line

Niyobuzima commented on src/routes/API/SendResetEmail.js:
```diff
@@ -0,0 +1,76 @@
+import express from 'express';
+import router from 'express'
```
delete this line it is not needed

dushimeemma reviewed (approved):
Good job 👍
